RadixWallet/Clients/AccountPortfoliosClient/AccountPortfoliosClient+State.swift

```diff
@@ -0,0 +1,285 @@
+	}
+
+	private func setOrUpdateAccountPortfolio(_ portfolio: AccountPortfoliosClient.AccountPortfolio) {
+		portfoliosSubject.value.mutateValue {
+			$0.updateValue(portfolio, forKey: portfolio.account.address)
+		}
+	}
+
+	private func setOrUpdateAccountPortfolios(_ portfolios: [AccountPortfoliosClient.AccountPortfolio]) {
+		var newValue: [AccountAddress: AccountPortfoliosClient.AccountPortfolio] = [:]
+		for portfolio in portfolios {
+			newValue[portfolio.account.address] = portfolio
+		}
+		portfoliosSubject.value = .success(newValue)
+	}
+}
+
+// MARK: - Fiat worth setters
+extension AccountPortfoliosClient.State {
+	func applyFiatWorth(_ portfolio: inout AccountPortfoliosClient.AccountPortfolio) {
+		applyTokenPrices(to: &portfolio)
+		applyCurrencyVisibility(to: &portfolio)
+		applyFiatCurrency(to: &portfolio)
+	}
+
+	func setTokenPrices(_ tokenPrices: TokenPrices) {
+		self.tokenPrices = tokenPrices
+		if var existingPortfolios = portfoliosSubject.value.values.wrappedValue.map(Array.init) {
+			applyTokenPrices(to: &existingPortfolios)
+			setOrUpdateAccountPortfolios(existingPortfolios)
+		}
+	}
+
+	func setIsCurrencyAmountVisble(_ isVisible: Bool) {
+		self.isCurrencyAmountVisible = isVisible
+		if var existingPortfolios = portfoliosSubject.value.values.wrappedValue.map(Array.init) {
+			applyCurrencyVisibility(to: &existingPortfolios)
+			setOrUpdateAccountPortfolios(existingPortfolios)
+		}
+	}
+
+	func setSelectedCurrency(_ currency: FiatCurrency) {
+		self.selectedCurrency = currency
+		if var existingPortfolios = portfoliosSubject.value.values.wrappedValue.map(Array.init) {
+			applyFiatCurrency(to: &existingPortfolios)
+			setOrUpdateAccountPortfolios(existingPortfolios)
+		}
+	}
+
+	func applyTokenPrices(to portfolios: inout [AccountPortfoliosClient.AccountPortfolio]) {
+		portfolios.mutateAll(applyTokenPrices)
+	}
+
+	func applyTokenPrices(to portfolio: inout AccountPortfoliosClient.AccountPortfolio) {
+		portfolio.updateFiatWorth(calculateWorth(gateway))
+	}
+
+	func applyCurrencyVisibility(to portfolio: inout AccountPortfoliosClient.AccountPortfolio) {
+		portfolio.isCurrencyAmountVisible = isCurrencyAmountVisible
+		portfolio.updateFiatWorth(value: isCurrencyAmountVisible, to: \.isVisible)
+	}
+
+	func applyCurrencyVisibility(to portfolios: inout [AccountPortfoliosClient.AccountPortfolio]) {
+		portfolios.mutateAll(applyCurrencyVisibility)
+	}
+
+	func applyFiatCurrency(to portfolio: inout AccountPortfoliosClient.AccountPortfolio) {
+		portfolio.fiatCurrency = self.selectedCurrency
+		portfolio.updateFiatWorth(value: selectedCurrency, to: \.currency)
+	}
+
+	func applyFiatCurrency(to portfolios: inout [AccountPortfoliosClient.AccountPortfolio]) {
+		portfolios.mutateAll(applyFiatCurrency)
+	}
+}
+
+// MARK: - Stake and Pool details handling
+extension AccountPortfoliosClient.State {
+	func calculateWorth(_ gateway: Radix.Gateway) -> (ResourceAddress, ResourceAmount) -> FiatWorth? {
```

Comment:
Helper function that calculates the worth for a given resource address and resource amount.